pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -57,9 +68,10 @@ func (e TopologyAffinityError) Type() string {
 type Manager interface {
 	// PodAdmitHandler is implemented by Manager
 	lifecycle.PodAdmitHandler
-	// AddHintProvider adds a hint provider to manager to indicate the hint provider
-	// wants to be consulted with when making topology hints
-	AddHintProvider(HintProvider)
+	// RegisterProvider adds a hint provider to manager to indicate the hint provider
+	// wants to be consulted with when making topology hints; the given AllocationInfoProvider
+	// is authoritative about the current resource allocation.
+	RegisterProvider(h HintProvider, a AllocationInfoProvider)
```

Comment:
It's weird to me to have this function take two parameters if its the same underlying object always implementing both interfaces) . Either we should come up with a single, new abstraction that makes sense as a Hint/AllocationInfo provider together (because they always *must* be implemented by the same object), or two separate calls (in case some hint provider doesn't actually want to provide allocation info (though I don't see when that would ever be the case).

Reply:
Thanks, very valid point. I think the best option is to have a new abstraction for the very reason you mentioned. I'll think about something. Naming will be fun, I'm exploring on the lines of `ResourceAllocator`